src/PermissionedUSDCWrapper.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v5.0.0) (token/ERC20/extensions/ERC20Wrapper.sol)
+
+pragma solidity ^0.8.20;
+
+import {IERC20Metadata} from "./ERC20.sol";
+import {ERC20, ERC20Wrapper, ERC20Permit, IERC20} from "lib/erc20-permissioned/src/ERC20PermissionedBase.sol";
+import {SafeERC20} from "./SafeERC20.sol";
+import {Auth} from "./utils/Auth.sol";
+import {IERC20PermissionedBase} from "src/interfaces/IERC20PermissionedBase.sol";
+import {Memberlist} from "src/Memberlist.sol";
+
+/**
+ * @dev Extension of the ERC-20 token contract to support token wrapping.
+ *
+ * Users can deposit and withdraw "underlying tokens" and receive a matching number of "wrapped tokens". This is useful
+ * in conjunction with other modules. For example, combining this wrapping mechanism with {ERC20Votes} will allow the
+ * wrapping of an existing "basic" ERC-20 into a governance token.
+ *
+ * WARNING: Any mechanism in which the underlying token changes the {balanceOf} of an account without an explicit transfer
+ * may desynchronize this contract's supply and its underlying balance. Please exercise caution when wrapping tokens that
+ * may undercollateralize the wrapper (i.e. wrapper's total supply is higher than its underlying balance). See {_recover}
+ * for recovering value accrued to the wrapper.
+ */
+
```

Comment:
nit: would be good to update this comment to describe the functionality of the attestations etc, rather than this old comment from the wrapped erc20 contract